### ISSUE TITLE
Crash when "Export to PDF"

### ISSUE THREAD

**Issue by javil33t** [giuspen/cherrytree#1291]
**Version, Operative system**
Ubuntu 20.04, cherrytree 0.99.13

**Describe the bug**
Cherrytree crash when 'Export to PDF' a node that has a code block bigger than the size of the A4 page. The bug is also triggered when the basic block needs to be splitted because the final of the page is reached 

**To Reproduce**
1. paste the following snippet in a code blocks
```
    

app = Flask(__name__)


#a
#boom


mail = Mail(app)
DATABASE = '/home/faraday/web/database.db'

template = '''
Se reporto un evento en SERVER:
{{ mensaje }}
'''

def get_db():
 db.row_factory = sqlite3.Row
 db.text_factory = lambda x: str(x).replace('{{', '')
return db

@app.teardown_appcontext
def close_connection(exception):
    db = getattr(g, '_database', None)
    if db is not None:
        db.close()

def init_db():
    with app.app_context():
        db = get_db()
        with app.open_resource('schema.sql', mode='r') as f:
           db.cursor().executescript(f.read())
        db.commit()

def query_db(query, args=(), one=False):
    with app.app_context():
        cur = get_db().execute(query, args)
        rv = [dict((cur.description[idx][0], str(value)) \
                for idx, value in enumerate(row)) for row in cur.fetchall()]
        return (rv[0] if rv else None) if one else rv

@app.route('/sendMessage', methods=['POST'])
def sendMessage():
    if request.method == 'POST':
        msg = Message(request.form['subject'], sender = 'faradaysec@gmail.com', recipients = [request.form['dest']])
        msg.body = render_template_string(template.replace('SERVER', query_db('SELECT nombre FROM usuarios ORDER BY usuario_id DESC', one=True)['nombre']), mensaje=request.form['body'])
        mail.send(msg)

        return render_template('enviado.html', dest=request.form['dest'])

@app.route('/profile')
def profile():
    name = request.args.get('name', '')
    if name:
        query_db('INSERT INTO usuarios (nombre) VALUES ("%s")' % name)

        return render_template('sender.html')
    return render_template('error.html')

@app.route('/')
def index():
    return render_template('base.html')

@app.route('/dump')
def dump():
    return Response(open(__file__).read(), mimetype='text/plain')

if __name__ == "__main__":
    app.run(host='0.0.0.0', port=8000)
```
2. Export to pdf
3. if the export was success, then add 2 comments more:
```
#a
#boom
#b
#c
```
4. Export and the process will be killed by a int3 in glib

**Screenshots**
![Selection_051](https://user-images.githubusercontent.com/72886737/97472990-334d7580-1929-11eb-8115-e3649d3f6b9e.png)

the exported version will be:
![Selection_052](https://user-images.githubusercontent.com/72886737/97473125-642daa80-1929-11eb-8c86-d4f659765d83.png)

add one line wherever you want and export again

![Selection_054](https://user-images.githubusercontent.com/72886737/97473351-a525bf00-1929-11eb-8425-af7eeae5f0cf.png)

then, the process trigger an exception
![Selection_055](https://user-images.githubusercontent.com/72886737/97473520-d4d4c700-1929-11eb-818e-a1007761368e.png)

this is the call stack
![Selection_056](https://user-images.githubusercontent.com/72886737/97473582-ecac4b00-1929-11eb-8db0-c54379689752.png)

the trigger is gonna be attached
[crash-cherrytree.zip](https://github.com/giuspen/cherrytree/files/5453795/crash-cherrytree.zip)

**Comment by giuspen** [comment]
Thanks this has been fixed already, soon in 0.99.18

**Comment by giuspen** [comment]
the fix was specifically in commit 47c53652 - thanks for the detailed bug report though!

**Comment by javil33t** [comment]
thanks to you for this amazing tool, I love it!
